tests/ui/traits/negative-impls/pin-unsound-issue-85099-derefmut.rs

```diff
@@ -0,0 +1,68 @@
+// check-pass
```

Comment:
Agh just saw that this test is not in the right place. Let's just put it under `tests/ui/typeck` or `tests/ui/issues`. (`tests/ui/deref` would be nicer, but not worth creating a whole directory for only one test)

Reply:
No worries, moved to `tests/ui/typeck`.